Lotus Daemon Exits When IPFS Integration Is Configured

**Describe the bug**
After configuring the IPFS integration per the docs [here](https://docs.lotu.sh/en+ipfs-client-integration), my lotus daemon exits with the following stack trace: 

```
2020-07-20T16:30:07.677-0500	WARN	modules	modules/services.go:44	failed to say hello	{"error": "protocol not supported", "peer": "12D3KooWHUugwRhzp9ZBMEnoCKoudRut9P7F2HqVbC4UJbZHq2Aw"}
panic: nyi

goroutine 2976 [running]:
github.com/filecoin-project/lotus/lib/bufbstore.(*BufferedBS).GetSize(0xc04ee49ac0, 0xc0fa0adb60, 0x22, 0x5, 0x1, 0xc09b713300)
	/Users/polluterofminds/Development/pinata/lotus/lib/bufbstore/buf_bstore.go:107 +0x39
github.com/ipfs/go-bitswap/internal/decision.(*blockstoreManager).getBlockSizes.func1(0xc0fa0adb60, 0x22)
	/Users/polluterofminds/go/pkg/mod/github.com/ipfs/go-bitswap@v0.2.8/internal/decision/blockstoremanager.go:73 +0x70
github.com/ipfs/go-bitswap/internal/decision.(*blockstoreManager).jobPerKey.func1()
	/Users/polluterofminds/go/pkg/mod/github.com/ipfs/go-bitswap@v0.2.8/internal/decision/blockstoremanager.go:116 +0x40
github.com/ipfs/go-bitswap/internal/decision.(*blockstoreManager).worker(0xc09dfa1230)
	/Users/polluterofminds/go/pkg/mod/github.com/ipfs/go-bitswap@v0.2.8/internal/decision/blockstoremanager.go:49 +0x3b
github.com/ipfs/go-bitswap/internal/decision.(*blockstoreManager).start.func1(0x6547760, 0xc09e010780)
	/Users/polluterofminds/go/pkg/mod/github.com/ipfs/go-bitswap@v0.2.8/internal/decision/blockstoremanager.go:38 +0x2a
github.com/jbenet/goprocess.(*process).Go.func1(0xc09dd77790, 0xc09e010780, 0xc09e0107e0)
	/Users/polluterofminds/go/pkg/mod/github.com/jbenet/goprocess@v0.1.4/impl-mutex.go:134 +0x3c
created by github.com/jbenet/goprocess.(*process).Go
	/Users/polluterofminds/go/pkg/mod/github.com/jbenet/goprocess@v0.1.4/impl-mutex.go:133 +0x1f6
```

If I remove the `UseIPFS` flag from the `config.toml` file, the daemon starts and runs as expected. 

**To Reproduce**
Steps to reproduce the behavior:
1. Run lotus against current (as of opening this issue) master branch of louts (SHA - 9b1d2f4c137433c1fe1f5ed44ba2e70105a409f0)
2. Update the `config.toml` file and add: 
```
[Client]
  UseIpfs = true
```

**Expected behavior**
Lotus daemon starts, syncs, and continues to run as expected while providing the ability to store files using IPFS hashes from the local IPFS node I'm running. 


**Version (run `lotus version`):**

`lotus version 0.4.1+git.9b1d2f4c`

**Additional context**
[This Slack](https://filecoinproject.slack.com/archives/CPFTWMY7N/p1595276837225700) conversation may be helpful.

--- a/lib/bufbstore/buf_bstore.go
+++ b/lib/bufbstore/buf_bstore.go
@@ -109,7 +109,7 @@ func (bs *BufferedBS) GetSize(c cid.Cid) (int, error) {
 		return bs.write.GetSize(c)
 	}
 
-	return 0, err
+	return s, err
 }
 
 func (bs *BufferedBS) Put(blk block.Block) error {

--- a/lib/bufbstore/buf_bstore.go
+++ b/lib/bufbstore/buf_bstore.go
@@ -104,7 +104,12 @@ func (bs *BufferedBS) Get(c cid.Cid) (block.Block, error) {
 }
 
 func (bs *BufferedBS) GetSize(c cid.Cid) (int, error) {
-	panic("nyi")
+	s, err := bs.read.GetSize(c)
+	if err == bstore.ErrNotFound || s == 0 {
+		return bs.write.GetSize(c)
+	}
+
+	return 0, err
 }
 
 func (bs *BufferedBS) Put(blk block.Block) error {

